projects/extension/src/background/ConnectionManagerWithHealth.ts

```diff
@@ -289,9 +296,11 @@ export class ConnectionManagerWithHealth<SandboxId> {
               const result: { peers: number } = jsonRpcMessage.result
               chain.peers = result.peers
 
-              // Notify the `allChainsChangedCallbacks`.
-              this.#allChainsChangedCallbacks.forEach((cb) => cb())
-              this.#allChainsChangedCallbacks = []
+              // Notify of the change in status.
+              return {
+                origin: "connection-manager",
+                type: "chains-status-changed",
+              }
```

Comment:
Add a const object with the interface and reuse it instead of copy/pasting the object:

 ```tsx
              const statusChanged: ChainsStatusChanged = {
                origin: "connection-manager",
                type: "chains-status-changed",
              }
              
              ...............
              
              return statusChanged
```

Reply:
I don't see the advantage of doing that?

Reply:
Reduce copy/pasting

Reply:
But copy-pasting code is not a problem by itself.
It can cause problems in some situations, because you can modify the copy-pasted code at one place and not at another, but this doesn't apply here because TypeScript would detect it.